js/src/redux/providers/certifications/middleware.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { bytesToHex, hex2Ascii } from '../../../api/util/format';
+import { addCertification } from './actions';
+
+import badgeRegABI from '../../../contracts/abi/badgereg.json';
+import certifierABI from '../../../contracts/abi/certifier.json';
+
+const badgeRegAddress = '0xcF5A62987294fd2087252FD812443508528C52bF';
+const ZERO = '0x0000000000000000000000000000000000000000000000000000000000000000';
+const knownCertifiers = [ 'smsverification' ];
+
+export const fetchCertifier = (api) => {
+  const registry = api.newContract(badgeRegABI, badgeRegAddress);
```

Comment:
Why not having this in a contract JS file like `tokenreg`, `githubhint`, etc...?